Make changes to improve accessibility

p-galligan opened the pull request:
No major errors in accessibility as far as I can tell going down the checklist and using tools.

We _are_ still running into warnings, specifically in using the `<i>` tag for social media icons. Automated checkers suggest that this is bad practice, but I haven't tackled that because it seems part of the larger issue of social media icons.

Fixes #13

Commit:
add title and h1

adds title and h1 to search results page

Commit:
add index title

add title to index page

Commit:
change landmarks

Remove unnecessary landmark in footer.

Wrap content in item details page in a div with a main role to include it in landmarks.

Commit:
adding more main roles

Add main roles to home page and search results

Commit:
add skip links

Adds skip links and main sections to individual pages

Commit:
add main to 404

Add main to 404 page for skip link.

HaSistrunk commented:
Great! Could you help me with what scripts to run or things to do to get this site running locally?

Commit:
add page title

Add page title for 404 page

p-galligan commented:
Sure!

From your base project directory, run `npm install`. This installs node_modules.

From your base project directory, run `bundle exec jekyll serve`. This will launch the site (reachable at localhost:4000).

In another terminal window, from your base project directory run `node js/create-index.js`. This will make a basic search index on the sample data. You can search for things like "board," "york," etc.

HaSistrunk commented on _includes/footer.html:
```diff
@@ -1,4 +1,4 @@
-<footer class="footer" role="contentinfo">
+<footer class="footer">
```
I know some automated a11y checkers flag `role="contentinfo"` as being redundant, but I'd recommend we keep this in because it's still useful for some older browsers and doesn't have any negative impacts.

HaSistrunk commented on js/search.js:
```diff
@@ -22,9 +22,9 @@ $(document).ready(function() {
     }
     $(".results__list").prepend(`
-      <p class="results__summary">
+      <h1><p class="results__summary">
```
We can get rid of the `<p>` tag entirely on this, and just have it as `<h1 class="results__summary">...</h1>`

helrond replied:
The margins on `results__summary` will need some adjusting to override default h1 margins.

HaSistrunk reviewed (changes requested):
A few things I'm thinking:
1. In terms of heading structure, it might be nice to have the search results titles as h2s.
2. The focus styles on the search results tiles are a little funky. Can we do away with the browser default focus styles around the text, and just use the orange border hover style around the whole tile? Not a must, but might be nice.
3. In the modal, we should probably underline the link to "Login to Raccess" since color is the only indication that it is a link.
4. The datepicker is funky and seems to have accessibility issues with keyboard controls on older/non-chrome browsers. This is currently an open issue in dimes, so maybe we can take care of it there and pull it here when we have a good solution? https://github.com/RockefellerArchiveCenter/dimes/issues/80. It looks like this on my slightly old Chrome 
![image](https://user-images.githubusercontent.com/15254041/91882411-eb73e000-ec50-11ea-8974-1985cba1236a.png)

HaSistrunk committed:
add semicolons

helrond reviewed (changes requested):
A couple of global things:
- I'd move the `main` element up into the default template, since the pages that have that element use that layout.
- It looks like we need to update the skip link styles in DIMES-TNG and then roll them back into the code here. Can you file an issue here and assign to me?
- I'm wondering if the modal should have focus on the date input on load, rather than the close button? Does that sound right, @HaSistrunk?

p-galligan commented:
@helrond Re: including `main` in the default template: I _can_ do it and even had it that way at first. My only qualm about it was that splitting it up out of the default gave more control over where on the page it would jump to. Basically just a little more fine control.

helrond commented:
@p-galligan I mean you've looked at this more closely than I have, but it looks like `main` is the outer div of all of the layouts that use it, which implies to me that you could move it up one level in the templating system.

HaSistrunk commented:
> I'm wondering if the modal should have focus on the date input on load, rather than the close button? Does that sound right, @HaSistrunk?

That's a good question. Looked into this, and I think it should be set to the modal title to allow screen reader users to be able to read through the content, rather than landing on a button without context

Commit:
move main to default

Moves main to default layout.

Commit:
Merge branch 'accessibility' of https://github.com/RockefellerArchiveCenter/library_discovery into accessibility

Commit:
remove focus outline on tiles

Removes focus outline on tile links

p-galligan commented:
> > I'm wondering if the modal should have focus on the date input on load, rather than the close button? Does that sound right, @HaSistrunk?
> 
> That's a good question. Looked into this, and I think it should be set to the modal title to allow screen reader users to be able to read through the content, rather than landing on a button or input without context

@HaSistrunk Clarification on this. Do you mean the title of the entire modal `modal-header__title` or the object title?

HaSistrunk commented:
> > > I'm wondering if the modal should have focus on the date input on load, rather than the close button? Does that sound right, @HaSistrunk?
> > 
> > 
> > That's a good question. Looked into this, and I think it should be set to the modal title to allow screen reader users to be able to read through the content, rather than landing on a button or input without context
> 
> @HaSistrunk Clarification on this. Do you mean the title of the entire modal `modal-header__title` or the object title?

Sorry, I mean `<h2 class="modal-header__title">Request in Reading Room</h2>`

Commit:
Update modal styling

Makes `modal-header__title` the default focus and adds underlining to the "login to RACcess" button.

Commit:
tweak results__summary

Tweaks `results__summary` margins to not get

p-galligan commented:
Ok, @helrond  @HaSistrunk I've managed to cover all of the suggestions (besides the datepicker) except for changing the results tiles to `h2` headings. I just have not figured out how to get those to work.

helrond committed:
use h2s in search results

p-galligan commented:
Love it when Travis builds fail because Twitter is down. 🤣

helrond committed:
add quotes around item title in modal

helrond reviewed (approved):
Looks good to me!

HaSistrunk commented:
Yeah, nice!

p-galligan committed:
Merge branch 'development' into accessibility
